pkg/graph/variable/variable_test.go

```diff
@@ -0,0 +1,275 @@
+			standalone:     true,
+		},
+		{
+			name: "field descriptor with multiple expressions",
+			fieldDesc: FieldDescriptor{
+				Path:                 "spec.name",
+				Expressions:          []string{"${prefix}", "${suffix}"},
+				ExpectedTypes:        []string{"string"},
+				ExpectedSchema:       nil,
+				StandaloneExpression: false,
+			},
+			expectedPath:   "spec.name",
+			expectedExprs:  []string{"${prefix}", "${suffix}"},
+			expectedTypes:  []string{"string"},
+			expectedSchema: nil,
+			standalone:     false,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			assert.Equal(t, tc.expectedPath, tc.fieldDesc.Path)
+			assert.Equal(t, tc.expectedExprs, tc.fieldDesc.Expressions)
+			assert.Equal(t, tc.expectedTypes, tc.fieldDesc.ExpectedTypes)
+			assert.Equal(t, tc.expectedSchema, tc.fieldDesc.ExpectedSchema)
+			assert.Equal(t, tc.standalone, tc.fieldDesc.StandaloneExpression)
+		})
+	}
+}
+
+func TestResourceField(t *testing.T) {
+	tests := []struct {
+		name          string
+		resourceField ResourceField
+		expectedKind  ResourceVariableKind
+		expectedDeps  []string
+	}{
+		{
+			name: "static resource field",
+			resourceField: ResourceField{
+				FieldDescriptor: FieldDescriptor{
+					Path:        "spec.replicas",
+					Expressions: []string{"${schema.spec.replicas + 5}"},
+				},
+				Kind:         ResourceVariableKindStatic,
+				Dependencies: []string{"instance"},
+			},
+			expectedKind: ResourceVariableKindStatic,
+			expectedDeps: []string{"instance"},
+		},
+		{
+			name: "dynamic resource field",
+			resourceField: ResourceField{
+				FieldDescriptor: FieldDescriptor{
+					Path:        "spec.vpcId",
+					Expressions: []string{"${vpc.status.vpcId}"},
+				},
+				Kind:         ResourceVariableKindDynamic,
+				Dependencies: []string{"vpc"},
+			},
+			expectedKind: ResourceVariableKindDynamic,
+			expectedDeps: []string{"vpc"},
+		},
+		{
+			name: "readyWhen resource field",
+			resourceField: ResourceField{
+				FieldDescriptor: FieldDescriptor{
+					Path:        "readyWhen",
+					Expressions: []string{"${cluster.status.status == \"Active\"}"},
+				},
+				Kind:         ResourceVariableKindReadyWhen,
+				Dependencies: []string{"cluster"},
+			},
+			expectedKind: ResourceVariableKindReadyWhen,
+			expectedDeps: []string{"cluster"},
+		},
+		{
+			name: "includeWhen resource field",
+			resourceField: ResourceField{
+				FieldDescriptor: FieldDescriptor{
+					Path:        "includeWhen",
+					Expressions: []string{"${schema.spec.replicas > 1}"},
+				},
+				Kind:         ResourceVariableKindIncludeWhen,
+				Dependencies: []string{"instance"},
+			},
+			expectedKind: ResourceVariableKindIncludeWhen,
+			expectedDeps: []string{"instance"},
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			assert.Equal(t, tc.expectedKind, tc.resourceField.Kind)
+			assert.Equal(t, tc.expectedDeps, tc.resourceField.Dependencies)
+		})
+	}
```

Comment:
Not sure if this is testing anything really, it's just assert things inside `tc` without calling any function